signac/synced_collections/buffers/serialized_file_buffered_collection.py

```diff
@@ -0,0 +1,309 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Buffering for file-based backends using a serialized buffer.
+
+The buffering method implemented here involves a single buffer of serialized
+data. All collections in buffered mode encode their data into this buffer on save
+and decode from it on load.
+"""
+
+import hashlib
+import json
+
+from ..errors import MetadataError
+from ..utils import SyncedCollectionJSONEncoder
+from .file_buffered_collection import FileBufferedCollection
+
+
+class SerializedFileBufferedCollection(FileBufferedCollection):
```

Comment:
Are we going to use the shared memory buffer (instead of the serialized file buffer) in signac 1.7, signac 2.0, or is this undecided yet?

Reply:
I hadn't decided on a specific timeline. Definitely by signac 2.0, I just wanted to get this pull request merged and get some testing in the wild before switching that on master. Depending on how long we wait until 1.7, this could be either in 1.7 or in 2.0.